pkg/generic/generic_service.go

```diff
@@ -37,21 +37,35 @@ type Service interface {
 
 // ServiceInfoWithCodec create a generic ServiceInfo with CodecInfo
 func ServiceInfoWithCodec(g Generic) *serviceinfo.ServiceInfo {
-	return newServiceInfo(g.PayloadCodecType(), g.MessageReaderWriter(), g.IDLServiceName())
+	return newServiceInfo(g.PayloadCodecType(), g.MessageReaderWriter(), g.IDLServiceName(), true)
 }
 
 // Deprecated: it's not used by kitex anymore.
 // ServiceInfo create a generic ServiceInfo
 func ServiceInfo(pcType serviceinfo.PayloadCodec) *serviceinfo.ServiceInfo {
-	return newServiceInfo(pcType, nil, "")
+	return newServiceInfo(pcType, nil, "", false)
 }
 
-func newServiceInfo(pcType serviceinfo.PayloadCodec, messageReaderWriter interface{}, serviceName string) *serviceinfo.ServiceInfo {
+func newServiceInfo(pcType serviceinfo.PayloadCodec, messageReaderWriter interface{}, serviceName string, withCodec bool) *serviceinfo.ServiceInfo {
 	handlerType := (*Service)(nil)
 
-	var methods map[string]serviceinfo.MethodInfo
-	var svcName string
+	methods, svcName := GetMethodInfo(messageReaderWriter, serviceName)
 
+	svcInfo := &serviceinfo.ServiceInfo{
+		ServiceName:  svcName,
+		HandlerType:  handlerType,
+		Methods:      methods,
+		PayloadCodec: pcType,
+		Extra:        make(map[string]interface{}),
+	}
+	svcInfo.Extra["generic"] = true
+	if !withCodec {
+		svcInfo.Extra["deprecated"] = true
+	}
+	return svcInfo
+}
+
+func GetMethodInfo(messageReaderWriter interface{}, serviceName string) (methods map[string]serviceinfo.MethodInfo, svcName string) {
```

Comment:
// GetMethodInfo is only used in kitex, please **DON'T USE IT**. This method may be removed in the future